tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/DataManagement/Courses/ErrorsTests.cs

```diff
@@ -0,0 +1,215 @@
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.Models;
+using Dfc.CourseDirectory.Core.Validation;
+using Dfc.CourseDirectory.Testing;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Dfc.CourseDirectory.WebV2.Tests.FeatureTests.DataManagement.Courses
+{
+    public class ErrorsTests : MvcTestBase
+    {
+        public ErrorsTests(CourseDirectoryApplicationFactory factory)
+            : base(factory)
+        {
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(UploadStatus.Created)]
+        [InlineData(UploadStatus.Processing)]
+        [InlineData(UploadStatus.Published)]
+        [InlineData(UploadStatus.Abandoned)]
+        public async Task Get_NoUnpublishedCourseUpload_ReturnsBadRequest(UploadStatus? uploadStatus)
+        {
+            // Arrange
+            var provider = await TestData.CreateProvider();
+
+            if (uploadStatus != null)
+            {
+                await TestData.CreateCourseUpload(provider.ProviderId, createdBy: User.ToUserInfo(), uploadStatus.Value);
+            }
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/data-upload/courses/errors?providerId={provider.ProviderId}");
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
+
+        [Fact]
+        public async Task Get_ValidRequest_RendersExpectedContent()
+        {
+            // Arrange
+            var provider = await TestData.CreateProvider();
+
+            var (_, courseUploadRows) = await TestData.CreateCourseUpload(
+                provider.ProviderId,
+                createdBy: User.ToUserInfo(),
+                UploadStatus.ProcessedWithErrors,
+                rowBuilder =>
+                {
+                    rowBuilder.AddValidRows(learnAimRef: string.Empty, 1);
```

Comment:
`learnAimRef` shouldn't be empty here; use `TestData.CreateLearningAimRef` to create one.